File: clients/java1.8_client/client/src/main/java/core/API/Passenger.java
package core.API;

import org.json.simple.JSONObject;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Boris on 01.02.17.
 */
public class Passenger implements MessagesInterface {
    private Integer id;
    private Integer elevator;
    private Integer fromFloor;
    private Integer destFloor;
    private Integer state;
    private Integer timeToAway;
    private String type;
    private Integer floor;
    private List<JSONObject> messages;
    private Double x;
    private Double y;
    private Double weight;

    public Boolean hasElevator() {
        return this.elevator != null;
    }

    public List<JSONObject> getMessages() {
        return this.messages;
    }

    public Integer getFromFloor() {
        return this.fromFloor;
    }

    public Integer getDestFloor() {
        return this.destFloor;
    }

    public Double getY() {
        return y;
    }

    public Integer getState() {
        return state;
    }

    public Double getX() {
        return x;
    }

    public Double getWeight() {
        return weight;
    }

    public Integer getElevator() {
        return elevator;
    }

    public Integer getId() {
        return id;
    }

    public Integer getTimeToAway() { return timeToAway; }

    public String getType() { return type; }

    public Integer getFloor() { return floor; }

    public Passenger(JSONObject passenger) {
        id = (int) (long) passenger.get("id");
        elevator = passenger.get("elevator") == null ? null : (int) (long) passenger.get("elevator");
        fromFloor = (int) (long) passenger.get("from_floor");
        destFloor = (int) (long) passenger.get("dest_floor");
        timeToAway = (int) (long) passenger.get("time_to_away");
        type = (String) passenger.get("type");
        floor = (int) (long) passenger.get("floor");
        state = (int) (long) passenger.get("state");
        messages = new ArrayList<>();
        if (passenger.get("x") instanceof Long) {
            x = ((Long) passenger.get("x")).doubleValue();
        } else {
            x = (double) passenger.get("x");
        }
        if (passenger.get("y") instanceof Long) {
            y = ((Long) passenger.get("y")).doubleValue();
        } else {
            y = (double) passenger.get("y");
        }

        if (passenger.get("weight") instanceof Long) {
            weight = ((Long) passenger.get("weight")).doubleValue();
        } else {
            weight = (double) passenger.get("weight");
        }
    }
    public void setElevator(Elevator elevator) {
        this.elevator = elevator.getId();
        JSONObject jo = new JSONObject();
        jo.put("command", "set_elevator_to_passenger");
        JSONObject args = new JSONObject();
        args.put("passenger_id", this.id);
        args.put("elevator_id", elevator.getId());
        jo.put("args", args);
        this.messages.add(jo);
    }

}

File: clients/kotlin_client/client/src/main/kotlin/core/API/Passenger.kt
package core.API

import org.json.simple.JSONObject

class Passenger(passenger: JSONObject) : MessagesInterface{

  val id: Int = passenger["id"]?.toString()?.toInt() ?: 0
  var elevator: Int = passenger["elevator"]?.toString()?.toInt() ?: 0
    private set

  val fromFloor: Int = passenger["from_floor"]?.toString()?.toInt() ?: 0
  val destFloor: Int = passenger["dest_floor"]?.toString()?.toInt() ?: 0
  val state: Int = passenger["state"]?.toString()?.toInt() ?: 0
  val timeToAway: Int = passenger["time_to_away"]?.toString()?.toInt() ?: 0
  val type: String = passenger["type"]?.toString() ?: ""
  val floor: Int = passenger["floor"]?.toString()?.toLong()?.toInt() ?: 0
  private val messages = ArrayList<JSONObject>()
  var x: Double = passenger["x"]?.toString()?.toDouble() ?: 0.0
  var y: Double = passenger["y"]?.toString()?.toDouble() ?: 0.0
  val weight: Double = passenger["weight"]?.toString()?.toDouble() ?: 1.0

  override fun getMessages(): List<JSONObject> = messages

  fun setElevator(elevator: Elevator) {
    this.elevator = elevator.id
    val jo = JSONObject()
    jo.put("command", "set_elevator_to_passenger")
    val args = JSONObject()
    args.put("passenger_id", this.id)
    args.put("elevator_id", elevator.id)
    jo.put("args", args)
    this.messages.add(jo)
  }
}